python/Ganga/GPIDev/Lib/File/MassStorageFile.py

```diff
@@ -238,39 +281,21 @@ def put(self):
 
         massStorageConfig = getConfig('Output')['MassStorageFile']['uploadOptions']
 
-        mkdir_cmd = massStorageConfig['mkdir_cmd']
         cp_cmd = massStorageConfig['cp_cmd']
         ls_cmd = massStorageConfig['ls_cmd']
         massStoragePath = massStorageConfig['path']
 
-        # create the last directory (if not exist) from the config path
-        import os.path
-        pathToDirName = os.path.dirname(massStoragePath)
-        dirName = os.path.basename(massStoragePath)
-
-        (exitcode, mystdout, mystderr) = self.execSyscmdSubprocess('%s %s' % (ls_cmd, pathToDirName))
-        if exitcode != 0:
-            self.handleUploadFailure(mystderr)
+        try:
+            self._mkdir(massStoragePath)
```

Comment:
This is now explicitly changing the behaviour from 'Check the parent dir exists, if not bail' to 'Create the whole dir structure as provided using `mkdir -p`'. I'm not saying this is necessarily a bad idea but it is a change. I'm not sure if anyone else has thoughts about this?